### PR TITLE
fix: changes to fix lifetime issues when multiple sdks are in use

### PR DESCRIPTION
**Pull request by kneekey23**
## Description of changes

This corrects a lifetime issue we had when multiple services are in use and attempting to use shared resources.
Corresponding PR: https://github.com/awslabs/aws-sdk-swift/pull/501

## Scope

- [ X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinboswell — APPROVED** [review]
You will definitely want this in your next release or anyone who tries to use multiple services simultaneously is gonna have a bad time

File: Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift [hunk 1]
```diff
@@ -9,12 +9,22 @@ import AwsCommonRuntimeKit
 import class Foundation.ProcessInfo
 
 public final class SDKDefaultIO {
-    public static let shared = SDKDefaultIO()
+    static weak var privateShared: SDKDefaultIO? = nil
```
Comment (wooj2) [inline Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift:12]: 1.  No need for `= nil` per the swift lint
2. Why is it weak?
3. Is this how we are suppose to do singletons? Seems like back in the day of obj-c we'd use dispatch_once ?

File: Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift [hunk 2]
```diff
@@ -9,12 +9,22 @@ import AwsCommonRuntimeKit
...
-    public let tlsContext: TlsContext
+    public static var shared: SDKDefaultIO {
+        if let shared = privateShared {
+            return shared
+        } else {
+            let shared = SDKDefaultIO()
+            privateShared = shared
+            return shared
+        }
+    }
```
Comment (wooj2) [inline Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift:22]: Is this how we are suppose to do singletons? Seems like back in the day of obj-c we'd use dispatch_once ?

File: Packages/ClientRuntime/Sources/Retries/SDKRetryer.swift [hunk 3]
```diff
@@ -8,6 +8,7 @@ import AwsCommonRuntimeKit
 
 public class SDKRetryer: Retryer {
     let crtRetryStrategy: CRTAWSRetryStrategy
+    private let sharedDefaultIO: SDKDefaultIO = SDKDefaultIO.shared
```
Comment (wooj2) [inline Packages/ClientRuntime/Sources/Retries/SDKRetryer.swift:11]: It is not entirely obvious where this new member variable is being used.

**Review by wooj2 — APPROVED** [review]
lgtm -- lets add a todo to verify thread safety

**Pull request merged**